File: app/src/main/java/com/hussain/podcastapp/model/Channel.java
package com.hussain.podcastapp.model;


import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Path;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.Text;

import java.util.ArrayList;

@Root(name = "channel", strict = false)
public class Channel {
    @ElementList(name = "item", inline = true)
    private ArrayList<Item> items;
    @Element(name = "title")
    private String title;
    @Element(name = "description")
    private String description;
    @Path("link")
    @Text(required = false)
    private String shareLink;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public ArrayList<Item> getItems() {
        return items;
    }

    public void setItems(ArrayList<Item> items) {
        this.items = items;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getShareLink() {
        return shareLink;
    }

    public void setShareLink(String shareLink) {
        this.shareLink = shareLink;
    }
}

File: app/src/main/java/com/hussain/podcastapp/model/Channel.kt
package com.hussain.podcastapp.model

import org.simpleframework.xml.*
import java.util.*

@Root(name = "channel", strict = false)
class Channel {
    @ElementList(name = "item", inline = true)
    var items: ArrayList<Item>? = null
    @Element(name = "title")
    var title: String? = null
    @Element(name = "description")
    var description: String? = null
    @Path("link")
    @Text(required = false)
    var shareLink: String? = null
}
